adoc/chapters/programming_interface.adoc

```diff
@@ -15580,45 +15567,65 @@ as a device copyable type in that case and the specialization is ignored.
-member variables encapsulated within the function object or variables captured by
-the lambda function must be treated according to the following rules:
-
-  * Any accessor must be passed as an argument to the device kernel in a
-    form that allows the device kernel to access the data in the specified
-    way.
-  * The <<sycl-runtime>> and compiler(s) must produce the necessary
-    conversions to enable accessor arguments from the host to be converted
-    to the correct type of parameter on the device.
-  * The device compiler(s) must validate that the layout of any data
-    shared between the host and the device(s) (e.g. value kernel arguments
-    or data accessed through an accessor or USM) is compatible with the
-    layout of that data on the host. If there is a layout mismatch that the
-    implementation cannot or will not correct for (to make the layouts
-    compatible), then the device compiler must issue an error and
-    compilation must fail.
-  * A local accessor provides access to work-group-local memory. The
-    accessor is not constructed with any buffer, but instead constructed
-    with a size and base data type. The runtime must ensure that the
-    work-group-local memory is allocated per work-group and available to be
-    used by the kernel via the local accessor.
-  * <<device-copyable,Device copyable>> types must be passed by value to the
-    kernel.
-  * Types that are not <<device-copyable>> must not be passed as arguments to a
-    kernel that is compiled for a device.
-  * It is illegal to pass a buffer or image (instead of an accessor
-    class) as an argument to a kernel. Generation of a compiler error in
-    this illegal case is optional.
-  * It is illegal to pass a reference argument to a kernel.
-    Generation of a compiler error in this illegal case is optional.
-  * If a pointer passed to a kernel is [code]#nullptr# or points to a <<usm>>
-    allocation, it must be passed through to the device unmodified. It is legal
-    to pass other pointers to a kernel, but dereferencing such pointers or
-    performing pointer arithmetic results in undefined behavior.
-  * Any aggregate types such as structs or classes should follow the rules
-    above recursively. It is not necessary to separate struct or class
-    members into separate kernel parameters if all members of the aggregate
-    type are unaffected by the rules above.
+A SYCL application passes parameters to a kernel in different ways depending on
+whether the kernel is a named function object or a lambda function.  If the
+kernel is a named function object, the [code]#operator()# member function (or
+other member functions that it calls) may reference member variables inside the
+same named function object.  Any such member variables become parameters to the
+kernel.  If the kernel is a lambda function, any variables captured by the
+lambda become parameters to the kernel.
```

Comment:
I didn't include these because I think this section is about kernel parameters that the *application* passes to a kernel.  The application never calls the `operator()` function directly.  Rather, the application calls something like `parallel_for`, and then the implementation calls `operator()`.

The list below is derived from the list I removed from section 4.4 "Class availability".  That list also did not include `item` as an allowed parameter type.

I think the question is whether the *application* can also pass an `item` as a parameter (e.g. by capturing a variable of type `item` in the lambda).  The answer to this is "no" because SYCL does not expose a constructor for `item`, so it is impossible for the application to create an object of this type.  The same goes for `nd_item`, `h_item`, `group`, and `subgroup`.

However, it is possible for an application to create an `nd_range` object, so I think it is an oversight that this type is missing from the list below.